pkg/ghactions/ghactions.go

```diff
@@ -114,15 +114,15 @@ func ModifyReferencesInYAMLWithCache(
 				continue
 			}
 
-			if shouldExclude(cfg, v.Value) {
-				continue
-			}
-
 			act, ref, err := ParseActionReference(v.Value)
 			if err != nil {
 				return modified, fmt.Errorf("failed to parse action reference '%s': %w", v.Value, err)
 			}
 
+			if shouldExclude(cfg, v.Value, act) {
+				continue
+			}
+
```

Comment:
Actually, after giving it some thought... what about having two exclude checks? One for the raw value + tag and another one for just the action name?

that would make the `shouldExclude` code simpler, but also improve performance (we don't do parsing if the action was already excluded).

Reply:
ha, I didn't think about it!